app/models/components/course/materials_ability_component.rb

```diff
@@ -41,8 +41,7 @@ def allow_students_upload_materials
   end
 
   def allow_staff_manage_materials
-    can :read, Course::Material, material_course_staff_hash
-    can :manage, Course::Material, material_course_staff_hash.deep_merge(concrete_material_hash)
+    can :manage, Course::Material, material_course_staff_hash
```

Comment:
Discussed offline, if we want different permissions for managers and TA, the permission need to be split.  Will defer that because we only know how to split when the requirements are clear.